internal/engine/compiler/engine.go

```diff
@@ -878,6 +882,7 @@ func (e *moduleEngine) newCallEngine(stackSize uint64, fn *function) *callEngine
 		archContext:   newArchContext(),
 		initialFn:     fn,
 		moduleContext: moduleContext{fn: fn},
+		stackIterator: &stackIterator{},
```

Comment:
super nit: can we avoid the allocation here by having stackIterator as a value, not ptr?